src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -44,16 +44,37 @@ export const initSafeSDK = async (
   chainId: string,
   safeAddress: string,
   safeVersion: string,
+  chainInfo: ChainInfo,
+  walletChainId: string,
 ): Promise<Safe> => {
   let isL1SafeMasterCopy = chainId === chains.eth
   // Legacy Safe contracts
   if (isLegacyVersion(safeVersion)) {
     isL1SafeMasterCopy = true
   }
 
-  const ethersProvider = new Web3Provider(provider)
+  const signerProvider = new Web3Provider(provider)
+
+  // Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
+  if (walletChainId !== chainId) {
+    const signerMethods = ['eth_sign', 'eth_signTypedData', 'eth_accounts', 'personal_sign']
+    const readonlyProvider = createWeb3ReadOnly(chainInfo.rpcUri)
+    const originalSend = signerProvider.send
+
+    signerProvider.send = (request, ...args) => {
+      console.log(request, ...args)
```

Comment:
Do we want to keep this on production?

Reply:
I'm not sure yet this entire hack can go to prod.